src/Nerdbank.Streams/MultiplexingStream.Channel.cs

```diff
@@ -832,13 +887,36 @@ private async Task ProcessOutboundTransmissionsAsync()
                 }
                 catch (Exception ex)
                 {
+                    string traceMessage = string.Empty;
                     if (ex is OperationCanceledException && this.DisposalToken.IsCancellationRequested)
                     {
                         await mxStreamIOReader!.CompleteAsync().ConfigureAwait(false);
+                        traceMessage = string.Format("Swallowed exception inside ProcessOutboundAsync: {0}", ex.Message);
                     }
                     else
                     {
                         await mxStreamIOReader!.CompleteAsync(ex).ConfigureAwait(false);
+
+                        // Record this as a faulting exception if the channel hasn't been disposed.
+                        lock (this.SyncObject)
+                        {
+                            if (!this.IsDisposed)
+                            {
+                                this.faultingException ??= ex;
+                                traceMessage = string.Format(
+                                    "Set faulting exception inside ProcessOutboundAsync: {0}",
+                                    this.faultingException.Message);
+                            }
+                        }
+                    }
+
+                    // Add a trace indicating that we caught an exception in ProcessOutbound
+                    if (this.TraceSource!.Switch.ShouldTrace(TraceEventType.Information))
+                    {
+                        this.TraceSource.TraceEvent(
+                            TraceEventType.Information,
```

Comment:
Using `Information` level messages to communicate a `TraceEventId.FatalError` seems ironic. But this isn't a fatal error. At least, IIRC, the FatalError code indicates the MultiplexingStream as a whole is going down, but it isn't. Just this channel is. Maybe we should define a new `ChannelFatalError` code? And then use an `TraceEventType.Error` level for it here.

Reply:
Does it make sense to introduce the ChannelFatalError code for just this one case. I am not sure what you generally consider a valid enough threshold for introducing a new code.